core/ir/synthetic/encode.c

```diff
@@ -41,92 +41,118 @@
 
 /* Encodes DR instruction representation \p instr into raw bytes \p encoded_instr.
  * Returns next instruction's pc.
- * A description of the encoding scheme is provided in core/ir/synthetic/decode.c.
- * We assume all encoded values to be little-endian.
+ * The encoding scheme followed is described in core/ir/synthetic/encoding_common.h.
  */
 byte *
 encode_to_synth(dcontext_t *dcontext, instr_t *instr, byte *encoded_instr)
 {
-    uint encoding = 0;
-    uint shift = 0;
+    /* Interpret the first 4 bytes of encoded_instr (which are always present) as a uint
+     * for easier retrieving of category, eflags, #src, and #dst values.
+     * We can do this safely because encoded_instr is 4 bytes aligned.
+     */
+    uint *encoding = ((uint *)&encoded_instr[0]);
+    *encoding = 0;
```

Comment:
Seems better to have a local value where all the bit operations are performed which is then written in one step.

Reply:
Yes, done.